http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -169,7 +172,7 @@ protected String getRedirectUri(String state) {
             for (String paramName : forwardableQueryParams) {
                 String paramValue = getQueryParamValue(facade, paramName);
                 if (SCOPE.equals(paramName)) {
-                    paramValue = addOidcScopeIfNeeded(paramValue);
+                    paramValue = addDeploymentAndQueryScopes(paramValue);
```

Comment:
To make the code a bit more clear, I think it would be good to handle scopes configured via the `OidcClientConfiguration` and query param configuration separately, i.e.,

Perhaps the helper method could be called `addScopes` and take the set of all scopes and a string as parameters. The helper method would then add the scopes from the string to the given set.

That way the new helper method could be called here to handle the query param specifically and could be called above to handle the `OidcClientConfiguration` scope value.